pkg/controller/certificates/sync.go

```diff
@@ -121,13 +142,44 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 
 	crtCopy := crt.DeepCopy()
 
+	// get issuer details for metrics
+	issuerName := issuerObj.GetObjectMeta().Name
+	var issuerScope string
+	switch crt.Spec.IssuerRef.Kind {
+	case v1alpha1.IssuerKind:
+		issuerScope = "namespaced"
+	case v1alpha1.ClusterIssuerKind:
+		issuerScope = "cluster"
+	default:
+		return fmt.Errorf(`invalid value %q for certificate issuer kind. Must be empty, %q or %q`, crt.Spec.IssuerRef.Kind, v1alpha1.IssuerKind, v1alpha1.ClusterIssuerKind)
+	}
+	var issuerType string
+	switch i.(type) {
+	case *acme.Acme:
+		issuerType = "acme"
+	case *ca.CA:
+		issuerType = "ca"
+	default:
+		issuerType = "unknown"
+	}
```

Comment:
We can avoid this switch on the type of `i` (which requires reflection).

There is a `nameForIssuer` function (https://github.com/jetstack/cert-manager/blob/master/pkg/issuer/const.go#L18) that we could export, which can be used in place here. Maybe rename it `IssuerType`? I want to explicitly avoid the word `kind` here, as the kind is the capitalised `ClusterIssuer` or `Issuer`.